src/backend/distributed/commands/table.c

```diff
@@ -155,6 +155,14 @@ ProcessCreateTableStmtPartitionOf(CreateStmt *createStatement)
 			char *parentRelationName = generate_qualified_relation_name(parentRelationId);
 			bool viaDeprecatedAPI = false;
 
+
+			/*
+			 * relation_access_tracking assumes we acquire locks on all partitions
+			 * of the parent relation.
+			 */
+			LockRelationOid(parentRelationId, ExclusiveLock);
+			LockRelationPartitions(parentRelationId, ExclusiveLock);
```

Comment:
This change looks surprising to me. We're getting here after `standardProcess_Utility` executed, right? So, doesn't Postgres already acquire the necessary relation locks? If not, are the locks really necessary?

Also, in what sense `relation_access_tracking ` requires the locks? Is it the regression tests of it requires or the actual tracking?